lib/caasp.pm

```diff
@@ -226,4 +226,24 @@ sub update_scheduled {
     }
 }
 
+# Repeat command until expected result or timeout
+# script_retry 'ping -c1 -W1 machine', retry => 5
+sub script_retry {
+    my ($cmd, %args) = @_;
+    my $ecode = $args{expect} // 0;
+    my $retry = $args{retry} // 10;
+    my $delay = $args{delay} // 30;
+
+    my $ret;
+    for (1 .. $retry) {
+        type_string "# Trying $_ of $retry:\n";
+
+        $ret = script_run($cmd);
+        last if defined($ret) && $ret == $ecode;
+
+        die("Waiting for Godot: $cmd") if $retry == $_;
```

Comment:
You are showing of your literacy? ;)

Reply:
I never read it, but seemed appropriate for script_retry :)

Reply:
hm, ok. I guess one has to live with line wrapping in the end. Anyway, shouldn't stop PR to be merged. Just be careful to not overdo it with record_info. I suggest to provide more details on fails but keep the information on passed jobs terse.